### PR TITLE
Fix subnet in docker_compose_net.yml

### PR DESCRIPTION
**Pull request by ilejn**
It seems that certain docker versions do not like 10.5.0.0/12 (which does not make sense indeed).
Probably this is the cause of test_dns_cache failures in https://s3.amazonaws.com/clickhouse-test-reports/0/a129d07eb58caa153f4ddae4ef60c033f94e5965/integration_tests__tsan__%5B3_6%5D.html

```
compose.cli.errors.log_api_error: invalid network config:
invalid subnet 10.5.0.0/12: it should be 10.0.0.0/12
```

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### CODE REVIEW

**Review by nikitamikhaylov — APPROVED** [review]

File: tests/integration/compose/docker_compose_net.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ networks:
     enable_ipv6: true
     ipam:
       config:
-      - subnet: 10.5.0.0/12
+      - subnet: 10.0.0.0/12
         gateway: 10.5.1.1
```
Comment (nikitamikhaylov) [inline tests/integration/compose/docker_compose_net.yml:9]:
```suggestion
        gateway: 10.0.0.1
```

Comment (tsolodov) [inline tests/integration/compose/docker_compose_net.yml:9]:
```suggestion
        gateway: 10.0.0.1
```

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 3ffb965b2330302e2c3b99b004dd9859208208ff with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69121/3ffb965b2330302e2c3b99b004dd9859208208ff/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/e7f403a27ee027cb117cca24167cb6d3629bba5e/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/e7f403a27ee027cb117cca24167cb6d3629bba5e/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/e7f403a27ee027cb117cca24167cb6d3629bba5e/stateless_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stress test</td><td>Runs stateless functional tests concurrently from several clients to detect concurrency-related errors</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/e7f403a27ee027cb117cca24167cb6d3629bba5e/stress_test__debug_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/950ca28f539eaac91bdaca5d80929b38f46c5a65/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by ilejn** [comment]
@nikitamikhaylov , why it was needed to change gateway? Any particular reason or just to make it more typical?
Asking out of curiosity.

**Pull request merged**